### DISCUSSION TITLE
Typescript support on next-auth@beta?

### DISCUSSION THREAD

**Discussion by denik1981** [nextauthjs/next-auth#3272] [Help] [answered]
Is Typescript support stable for next-auth@beta?
I note that the starter example repo for typescript isn't using beta but latest, and by adding typescript support to the beta example I came out with a bunch of type errors. Not sure what is the current state for typescript under beta or if it is advisable to use latest if we want to use TS.

**Comment by balazsorban44** [comment, marked as answer]
absolutely. there is a WIP PR that will be merged when v4 goes stable https://github.com/nextauthjs/next-auth-typescript-example/pull/8

Curious what errors you encountered. Might this be related? https://github.com/nextauthjs/next-auth/issues/2709

**Comment by denik1981** [comment]
Not sure if "errors" is the right term here. If you try adding typescript support to the nextjs app that runs the next-auth-example (beta no TS) and then rename the files from js to ts, the compiler will throw you a bunch of type errors. I started fixing a few of them but I was concern if I was doing a task that is already on a WIP. Thanks for letting me know.
